### PR TITLE
add docs to upgrade to Flux GA

### PR DESCRIPTION
**Pull request by enekofb**
**What changed?**

Added guide for wego users to upgrade to Flux GA

**Why was this change made?**

To support users in their upgrade journey

**How was this change implemented?**

Documentation

**How did you validate the change?**

Locally 

Screenshot 2023-05-24 at 09 51 58

### CODE REVIEW

File: website/docs/guides/fluxga-upgrade.mdx [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+At this stage you have Weave Gitops running Flux Beta.
+
+### 2. Upgrade to Flux GA in `ClusterBootstrapConfig`s
+
+First we ensure any new clusters are bootstrapped with Flux GA, then we'll upgrade the existing clusters.
+
+`ClusterBootstrapConfig` will most often contain an invocation of `flux bootstrap`, make sure the image is using `v2`.
+
+```patch
```
Comment (bigkevmcd) [inline website/docs/guides/fluxga-upgrade.mdx:140]: Could we trim this diff down a bit?

**Review by bigkevmcd — COMMENTED** [review]
Looks good, some tidyups before merging.

File: website/docs/guides/fluxga-upgrade.mdx [hunk 2]
```diff
@@ -0,0 +1,204 @@
+---
+title: Upgrade to Flux GA
+hide_title: true
+---
+
+# Upgrade to Flux GA
+
+We are very excited due Flux GA is being [released!](https://github.com/fluxcd/flux2/releases)
+
+This guide aims to answer some [common questions](#faq) before starting the upgrade, as well as, guiding how to do
```
[the remark this answers is not in the document]
Comment (enekofb) [reply website/docs/guides/fluxga-upgrade.mdx]: reworded 🙇

File: website/docs/guides/fluxga-upgrade.mdx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+## FAQ
+
+Here you could find the most common questions around upgrading.
+
+### Why upgrade to Flux GA
+
+Flux Beta APIs have been stable and used in production for quite some time now. However, with Flux GA there is big reason to move to v1:
+it is the main supported api version for new features and developments. Features like [horizontal scaling](https://fluxcd.io/flux/cheatsheets/sharding/).
+are only available in Flux GA. Also have into consideration that beta apis will be removed after 6 months.
```
Comment (Himangini) [inline website/docs/guides/fluxga-upgrade.mdx:39]:
```suggestion
are only available in Flux GA. Also, have into consideration that beta APIs will be removed after 6 months.
```

File: website/docs/guides/fluxga-upgrade.mdx [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+Use your regular WGE upgrade procedure to bring it to the [latest](https://docs.gitops.weave.works/docs/releases/)
+
+:::warning upgrading to Flux GA with WGE v0.22.0?
+
+Ensure that you address [this](#v0220)
+:::
+
+At this stage you have Weave Gitops running Flux Beta.
+
+### 2. Upgrade to Flux GA in `ClusterBootstrapConfig`s
```
Comment (Himangini) [inline website/docs/guides/fluxga-upgrade.mdx:136]:
```suggestion
### 2. Upgrade to Flux GA in ClusterBootstrapConfigs
```
to make consistent with point 4. below

**Review by Himangini — COMMENTED** [review]
small nits ✨

File: website/docs/guides/fluxga-upgrade.mdx [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+     generateName: "run-gitops-{{ .ObjectMeta.Name }}"
+     spec:
+       containers:
+-        - image: ghcr.io/fluxcd/flux-cli:v0.34.0
++        - image: ghcr.io/fluxcd/flux-cli:v2
+           name: flux-bootstrap
+           ...
+```
+</details>
+At this stage you new bootstrapped clusters will run Flux GA.
```
Comment (serboctor) [inline website/docs/guides/fluxga-upgrade.mdx:169]:
```suggestion
At this stage, your new bootstrapped clusters will run Flux GA.
```

File: website/docs/guides/fluxga-upgrade.mdx [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+### 3. Upgrade to Flux GA on your existing leaf clusters and management clusters
+
+Follow the upgrade instructions from the [Flux v2.0.0-rc.1 release notes](https://github.com/fluxcd/flux2/releases/tag/v2.0.0-rc.1)
```
Comment (serboctor) [inline website/docs/guides/fluxga-upgrade.mdx:173]:
```suggestion
Follow the upgrade instructions from the [Flux v2.0.0-rc.1 release notes](https://github.com/fluxcd/flux2/releases/tag/v2.0.0-rc.1).
```

File: website/docs/guides/fluxga-upgrade.mdx [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+
+At minimum, you'll need to rerun the `flux bootstrap` command on your leaf clusters and management clusters.
+
+You'll also need to bump api versions in your manifests to `v1` as described in the flux upgrade instructions:
```
Comment (serboctor) [inline website/docs/guides/fluxga-upgrade.mdx:177]:
```suggestion
You'll also need to bump APIs version in your manifests to `v1` as described in the flux upgrade instructions:
```

File: website/docs/guides/fluxga-upgrade.mdx [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+
+If you have updated your `Kustomization` and `GitRepository` resources to `v1` you will also be able to upgrade to the future release of Flux that will drop support for `< v1` apis.
+
+:::info
+
+Weave Gitops Enterprise 0.23.0 generates v1beta2 `Kustomization` Custom Resources when using the _Add app_ and _Cluster Template_ features.
+
+This is to support installations still using Flux Beta.
+
+After upgrading to the release of Weave Gitops that drops support for Flux Beta Weave Gitops Enterprise will create `v1` Resources.
```
Comment (serboctor) [inline website/docs/guides/fluxga-upgrade.mdx:208]:
```suggestion
After upgrading to the release of Weave Gitops that drops support for Flux Beta, Weave Gitops Enterprise will create `v1` Resources.
```

File: website/docs/guides/fluxga-upgrade.mdx [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+- [Azure AKS Flux-Gitops extension](https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/extensions-release#flux-gitops)
+
+As of writing they do not yet support the new version so please wait before upgrading to Flux GA.
+:::
+
+Below, we'll take you through the multiple steps required to migrate to your system to Flux GA. After each step the cluster will be
+in a working state, so you can take your time to complete the migration.
+
+1. Upgrade to [latest Weave Gitops](https://docs.gitops.weave.works/docs/releases/)
+2. Upgrade to Flux GA in `ClusterBootstrapConfig`s
```
[the remark this answers is not in the document]
Comment (enekofb) [reply website/docs/guides/fluxga-upgrade.mdx:121]: applied

**Review by serboctor — APPROVED** [review]
LGTM

**Review by Himangini — APPROVED** [review]
LGTM 👍🏻 nicely done ✨

**Review by foot — APPROVED** [review]
💯

**Comment by LappleApple** [comment]
Hiya, thanks for writing docs! One request and two fixes:
- please make sure all "Weave Gitops" and "Weave Gitops Enterprise" use the official product capitalization: Weave GitOps (Enterprise)
- "Also, have into consideration that beta APIs will be removed after 6 months." ==> "There's also the consideration that beta APIs will be removed after 6 months."
- "When support for Flux Beta will be dropped?" ==> "When will support for Flux Beta be dropped?"

**Pull request merged**